controllers/storagecluster/provider_server.go

```diff
@@ -145,6 +145,12 @@ func (o *ocsProviderServer) createDeployment(r *StorageClusterReconciler, instan
 
 func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
 
+	if instance.Spec.ProviderAPIServerServiceType == corev1.ServiceTypeClusterIP || instance.Spec.ProviderAPIServerServiceType == corev1.ServiceTypeExternalName {
```

Comment:
I believe we should check the services we currently support. This way, we can mitigate the risk associated with the potential addition of new services in the future. By focusing on the ones we already support, we can avoid encountering such problems.

```
if instance.Spec.ProviderAPIServerServiceType != "" &&
    instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeNodePort &&
    instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeLoadBalancer
```
It is just a suggestion and not a must, Let me know if you would like to do that, otherwise I will approve it.

Reply:
I will make this change and update the PR